testER/src/App.jsx

```diff
@@ -1,26 +1,30 @@
-import React, { useEffect, useState } from 'react'
+import React, { createContext, useEffect, useState } from 'react'
 import ReactGA from 'react-ga'
 import mapboxgl from 'mapbox-gl'
-import ReactDOM from 'react-dom'
-import SubjectPopup from './components/Popup'
-import Legend from './components/Legend.jsx'
+import SubjectPopupContent from './components/SubjectPopupContent'
+import Popup from './components/Popup'
+import Legend from './components/Legend'
 
 import './App.css'
 import 'mapbox-gl/dist/mapbox-gl.css'
 
 // instantiate the Map
 mapboxgl.accessToken = 'pk.eyJ1IjoidmpvZWxtIiwiYSI6ImNra2hiZXNpMzA1bTcybnA3OXlycnN2ZjcifQ.gH6Nls61WTMVutUH57jMJQ' // development token
-let GlobalMap
+
 let config
 const keymap = {} // alt, r
 
+window.GlobalMap = null
+
+export const TrackContext = createContext({})
+
 /* eslint-disable react/prop-types */
 const App = (props) => {
   var [subjects, setSubjects] = useState([])
   var [tracks, setTracks] = useState({})
   var [subjectColor, setSubjectColor] = useState({})
   var [legSub, setLegSub] = useState(undefined)
-  var [legSubAvailable, setLegSubAvailable] = useState({})
+  const [subjectPopups, setSubjectPopups] = useState([])
```

Comment:
Since we're now managing the visibility/existence of popups via React, you need a state container.